aiven/client/argx.py

```diff
@@ -196,47 +260,100 @@ def print_response(
+                        result_type=type(result)
+                    )
+                )
+            for table_row in prepared_data.table_rows:
+                output_stream.write(format.format(**table_row))
+                output_stream.write("\n")
 
+        if output_format in (OutputFormats.TABLE, OutputFormats.TABLE_NOHEADER):
             pretty.print_table(
                 result,
                 drop_fields=drop_fields,
                 table_layout=table_layout,
-                header=header,
-                file=file,
+                header=output_format == OutputFormats.TABLE,
+                file=output_stream
+            )
+
+        elif output_format in (OutputFormats.CSV, OutputFormats.CSV_NOHEADER):
+            prepared_data = pretty.prepare_table(result, table_layout)
+            fields = []
+            if len(prepared_data.vertical_fields) != 0:
+                fields.append("")
+
+            writer = csvlib.DictWriter(output_stream, extrasaction="ignore", fieldnames=fields)
+
+            if output_format == OutputFormats.CSV:
+                writer.writeheader()
+
+            for row in prepared_data.table_rows:
+                row_item = {k: pretty.format_item(v) for k, v in row.items()}
+                row_item[""] = "\n".join(pretty.yield_vertical_fields(row_item, prepared_data.vertical_fields))
+                writer.writerow(row_item)
+
+        elif output_format == OutputFormats.JSON:
+            jsonlib.dump(result, output_stream, indent=4, sort_keys=True, cls=pretty.CustomJsonEncoder)
+            output_stream.write("\n")
+        elif output_format == OutputFormats.JSON_COMPACT:
+            jsonlib.dump(result, output_stream, sort_keys=True, cls=pretty.CustomJsonEncoder)
+            output_stream.write("\n")
+        elif output_format == OutputFormats.YAML:
+            yamllib.round_trip_dump(result, output_stream, default_flow_style=False)
+
+        else:
+            raise NotImplementedError(
+                "BUG: unrecognized output format: {output_format}".format(output_format=output_format.value)
             )
 
     def run(self, args=None):
         args = args or sys.argv[1:]
         if not args:
             args = ["--help"]
 
+        # Parse the early arguments
+        early_args, _ = self.early_parser.parse_known_args(args)
+
+        # Configure log levels early
+        logging.basicConfig(
+            level=logging._nameToLevel[early_args.log_level],  # pylint: disable=protected-access
+            format=LOG_FORMAT
+        )
+        logging.getLogger("requests").setLevel(
+            logging._nameToLevel[early_args.request_log_level]  # pylint: disable=protected-access
+        )
+
+        self.config = Config(early_args.config)
+
+        self._output_format = early_args.output_format
+
+        if early_args.output == "-" or early_args.output == "":
```

Comment:
If `-` means stdout then might as well just call it stdout. I don't really see a reason to use a single dash to represent it unless I am missing something.

Reply:
Unix command line convention: `-` is the common specifier for stdout, whereas if you enter anything else then it's treated as a filename/filepath.

Reply:
This is true, but it doesn't mean that we need to adopt it here. I'd argue that `-` is less descriptive than `stdout` so might as well go with the latter.

Reply:
The problem is `stdout` is reasonably a file path, not stdout. A user at their console can actually ask bash to pass `/dev/stdout` if they must write literally what they want, but otherwise most common unix commands use this convention - 
```
$ curl -o - google.com
<HTML><HEAD><meta http-equiv="content-type" content="text/html;charset=utf-8">
<TITLE>301 Moved</TITLE></HEAD><BODY>
<H1>301 Moved</H1>
The document has moved
<A HREF="http://www.google.com/">here</A>.
</BODY></HTML>

$ wget -O - google.com
# truncated since it follows links
```